### ISSUE TITLE
Exception while executing meta query on MySQL 5.6: Unknown column 'information_schema.COLUMNS.GENERATION_EXPRESSION'

### ISSUE THREAD

**Issue by christiangyor** [jOOQ/jOOQ#14598]
### Expected behavior

org.jooq.codegen.GenerationTool to run without exceptions

### Actual behavior

```sql
select
  information_schema.COLUMNS.ORDINAL_POSITION,
  information_schema.COLUMNS.COLUMN_NAME,
  information_schema.COLUMNS.COLUMN_COMMENT,
  information_schema.COLUMNS.COLUMN_TYPE,
  information_schema.COLUMNS.DATA_TYPE,
  information_schema.COLUMNS.IS_NULLABLE,
  information_schema.COLUMNS.COLUMN_DEFAULT,
  information_schema.COLUMNS.EXTRA,
  information_schema.COLUMNS.GENERATION_EXPRESSION,
  information_schema.COLUMNS.CHARACTER_MAXIMUM_LENGTH,
  coalesce(information_schema.COLUMNS.NUMERIC_PRECISION, information_schema.COLUMNS.DATETIME_PRECISION) as NUMERIC_PRECISION,
  information_schema.COLUMNS.NUMERIC_SCALE,
  information_schema.COLUMNS.EXTRA
from information_schema.COLUMNS
where (
  information_schema.COLUMNS.TABLE_SCHEMA in (
    'XX', 'XX'
  )
  and information_schema.COLUMNS.TABLE_NAME = 'XX'
)
order by information_schema.COLUMNS.ORDINAL_POSITION```


17:53:09 SEVERE Error while initialising type
org.jooq.exception.DataAccessException: SQL [select information_schema.COLUMNS.ORDINAL_POSITION, information_schema.COLUMNS.COLUMN_NAME, information_schema.COLUMNS.COLUMN_COMMENT, information_schema.COLUMNS.COLUMN_TYPE, information_schema.COLUMNS.DATA_TYPE, information_schema.COLUMNS.IS_NULLABLE, information_schema.COLUMNS.COLUMN_DEFAULT, information_schema.COLUMNS.EXTRA, information_schema.COLUMNS.GENERATION_EXPRESSION, information_schema.COLUMNS.CHARACTER_MAXIMUM_LENGTH, coalesce(information_schema.COLUMNS.NUMERIC_PRECISION, information_schema.COLUMNS.DATETIME_PRECISION) as NUMERIC_PRECISION, information_schema.COLUMNS.NUMERIC_SCALE, information_schema.COLUMNS.EXTRA from information_schema.COLUMNS where (information_schema.COLUMNS.TABLE_SCHEMA in (?, ?) and information_schema.COLUMNS.TABLE_NAME = ?) order by information_schema.COLUMNS.ORDINAL_POSITION]; Unknown column 'information_schema.COLUMNS.GENERATION_EXPRESSION' in 'field list'
	at org.jooq_3.16.13.MYSQL.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3104)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:670)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:354)
	at org.jooq.impl.AbstractResultQuery.fetch(AbstractResultQuery.java:284)
	at org.jooq.impl.SelectImpl.fetch(SelectImpl.java:2843)
	at org.jooq.impl.ResultQueryTrait.iterator(ResultQueryTrait.java:300)
	at org.jooq.meta.mysql.MySQLTableDefinition.getElements0(MySQLTableDefinition.java:90)
	at org.jooq.meta.AbstractElementContainerDefinition.getElements(AbstractElementContainerDefinition.java:93)
	at org.jooq.meta.AbstractTableDefinition.getColumns(AbstractTableDefinition.java:186)
	at org.jooq.codegen.JavaGenerator.generateTable(JavaGenerator.java:5357)
	at org.jooq.codegen.JavaGenerator.generateTables(JavaGenerator.java:5344)
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:612)
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:567)
	at org.jooq.codegen.JavaGenerator.generate0(JavaGenerator.java:482)
	at org.jooq.codegen.AbstractGenerator.generate(AbstractGenerator.java:182)
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:201)
	at org.jooq.codegen.GenerationTool.run0(GenerationTool.java:912)
	at org.jooq.codegen.GenerationTool.run(GenerationTool.java:239)
	at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:234)
	at org.jooq.codegen.GenerationTool.main(GenerationTool.java:206)
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'information_schema.COLUMNS.GENERATION_EXPRESSION' in 'field list'
	at jdk.internal.reflect.GeneratedConstructorAccessor6.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:408)
	at com.mysql.jdbc.Util.getInstance(Util.java:383)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1062)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:4226)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:4158)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2615)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2776)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2840)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2082)
	at com.mysql.jdbc.PreparedStatement.execute(PreparedStatement.java:1302)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219)
	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4382)
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:230)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:340)
	... 17 more


### Steps to reproduce the problem

java -classpath jooq-3.16.13.jar:\
jooq-meta-3.16.13.jar:\
jooq-codegen-3.16.13.jar:\
reactive-streams-1.0.3.jar:\
r2dbc-spi-0.9.0.RELEASE.jar:\
jakarta.xml.bind-api-3.0.0.jar:\
mysql-connector-java-5.1.31-bin.jar:. \
org.jooq.codegen.GenerationTool /XX.xml

### jOOQ Version

3.16.13 open source 

### Database product and version

MySQL 5.6.51

### Java Version

openjdk 11.0.16.1 2022-08-12

### OS Version

MACOS MONTEREY

### JDBC driver name and version (include name if unofficial driver)

mysql-connector-java-5.1.31-bin.jar

**Comment by lukaseder** [comment]
Thanks a lot for your report. Indeed, the `GENERATION_EXPRESSION` column was added in MySQL 5.7 only:
https://dev.mysql.com/doc/refman/5.7/en/information-schema-columns-table.html

I'll fix this right away.

**Comment by lukaseder** [comment]
MariaDB 10.2 has added the column as well, older versions don't have it yet:
https://mariadb.com/kb/en/information-schema-columns-table/

**Comment by lukaseder** [comment]
This was fixed already for MemSQL, in the past: #13854

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.18.0, 3.17.8 (https://github.com/jOOQ/jOOQ/issues/14601), and 3.16.14 (https://github.com/jOOQ/jOOQ/issues/14602).

Please note that the jOOQ Open Source Edition requires MySQL 8. For a list of supported MySQL versions per edition and jOOQ version, please refer to the following support matrix:
https://www.jooq.org/download/support-matrix

The fix was implemented to possibly work on the jOOQ Open Source Edition as well, but we do not integration test the jOOQ Open Source Edition on MySQL 5.6.

**Comment by christiangyor** [comment]
Thank you for your quick response. Your feedback is another reason to upgrade this MySQL server.
